runtime/interpreter/value.go

```diff
@@ -17782,6 +17783,68 @@ func (v *CompositeValue) RemoveTypeKey(
 	return v.RemoveMember(interpreter, locationRange, attachmentMemberName(attachmentType))
 }
 
+func (v *CompositeValue) Iterator(interpreter *Interpreter) ValueIterator {
+	staticType := v.StaticType(interpreter)
+
+	switch typ := staticType.(type) {
+	case InclusiveRangeStaticType:
+		return NewInclusiveRangeIterator(v, interpreter, typ)
+
+	default:
+		// Must be caught in the checker.
+		panic(errors.NewUnreachableError())
+	}
+}
+
+type InclusiveRangeIterator struct {
+	rangeValue   *CompositeValue
+	next         IntegerValue
+	stepNegative bool
+}
+
+var _ ValueIterator = &InclusiveRangeIterator{}
+
+func NewInclusiveRangeIterator(
+	v *CompositeValue,
+	interpreter *Interpreter,
+	typ InclusiveRangeStaticType,
+) *InclusiveRangeIterator {
+	startValue := v.GetField(interpreter, EmptyLocationRange, sema.InclusiveRangeTypeStartFieldName).(IntegerValue)
+
+	zeroValue := GetSmallIntegerValue(0, typ.ElementType)
+	stepValue := v.GetField(interpreter, EmptyLocationRange, sema.InclusiveRangeTypeStepFieldName).(IntegerValue)
+	stepNegative := stepValue.Less(interpreter, zeroValue, EmptyLocationRange)
+
+	return &InclusiveRangeIterator{
+		rangeValue:   v,
+		next:         startValue,
+		stepNegative: bool(stepNegative),
+	}
+}
+
+func (i *InclusiveRangeIterator) Next(interpreter *Interpreter) Value {
+	valueToReturn := i.next
+
+	end := getFieldAsIntegerValue(i.rangeValue, interpreter, EmptyLocationRange, sema.InclusiveRangeTypeEndFieldName)
```

Comment:
`end` also needs to be computed only once. Can move this to the constructor as well.